### DISCUSSION TITLE
Data coming in weird, or choppy

### DISCUSSION THREAD

**Discussion by cnf** [someweisguy/esp_dmx#157] [General]
Hi,

I am trying to get a project working on an M5 core, with the DMX module, and the DMX data coming in keeps jumping between the right values, and 0... also, data received isn't smooth, it receives a bunch of data, and then nothing, and then a bunch of data, then nothing.

I am running with `if (dmx_receive(DMX_NUM_1, &packet, 0)) {}`  because I am also listening for sACN packets if no DMX wired data comes in, and updating other parts of the setup.

I am asking here, because i'm not sure where to go next, or what specific question to ask, 

The same data (coming from QLC+) sent on sACN is working well and stable, on wired DMX it works ok sometimes though laggy, and completely random at other times.

Thanks!

**Comment by someweisguy** [comment]
Interesting. I haven't had issues with choppy input. I haven't personally written code that concurrently handles sACN and DMX so I am guessing that is part of the issue. 

Have you looked into FreeRTOS tasks at all? I would try using a FreeRTOS task to receive DMX data. This way, you can block the task when no DMX is received. This also may solve the choppiness problems.

**Reply by cnf** [reply]
I'll have a look at that. Though I have a check in my main loop to see if it is taking too long, and it never triggers:
```cpp
if (enter + 50 < millis()) {
    log_w("We are taking too long! %d", millis() - enter);
  };
```

Would this also explain why the values jump between the actual channel value, and 0, back and forth?

I have put my dmx code [here](https://gist.github.com/cnf/13d50df09ce7f4e6b28e1435bfff477e) if you want to have a look.

Thanks!

**Reply by cnf** [reply]
I did a stupid and changed some things around, and now I can't replicate my own issue 😄 
I'll do some code refactoring, and some profiling to check the timings etc, and I'll report back if I can either replicate it again, or if it is "magically" gone.

This tells me it's mostly likely something I did, (like duh 😃 ) I just don't know what !

Thanks!
